File: java-solid-principle-sample-code/src/com/principle/solid/openclosed/ShapePrinter.java
package com.principle.solid.openclosed;

import java.util.List;

public class ShapePrinter {

    private AreaCalculator areaCalculator = null;

    public ShapePrinter(AreaCalculator areaCalculator) {
        this.areaCalculator = areaCalculator;
    }

    public String json(List<Shape> shapes) {
        return String.format("{shapes_sum: %s}", areaCalculator.sum(shapes));
    }

    public String csv(List<Shape> shapes) {
        return String.format("shapes_sum,%s", areaCalculator.sum(shapes));
    }
}

File: python-solid-principle-sample-code/com/principle/solid/single_responsibility/ShapePrinter.py
from AreaCalculator import AreaCalculator


class ShapePrinter:

    def __init__(self, area_calculator: AreaCalculator):
        self.__area_calculator = area_calculator

    def json(self, shapes):
        return "{{shapes_sum: {}}}".format(int(self.__area_calculator.sum(shapes)))

    def csv(self, shapes):
        return "shapes_sum,{}".format(int(self.__area_calculator.sum(shapes)))